* Path ./package/usr/lib/libcryptopp.so.8 is owned by uid 1111, gid 1111, which doesn't match any user/group on target. This may be due to host contamination.
@ 2021-05-07  8:34 Thomas Hill
  0 siblings, 0 replies; only message in thread
From: Thomas Hill @ 2021-05-07  8:34 UTC (permalink / raw)
  To: Yocto-mailing-list

[-- Attachment #1: Type: text/plain, Size: 2029 bytes --]

Hi!

I try to create a libcryptopp package for my yocto build with the appended bb-recipe.
That worked without problems on my old build server with Yocto 3.0.

Now I have moved to a new build server and upgrade to Yocto 3.3. Now I have the problem that the do_package function failed.

,---[ bitbake core-image-minimal ]---
| WARNING: libcryptopp-8.2.0+gitAUTOINC+9dcc26c582-r8 do_package: KeyError in ./package/usr/lib/libcryptopp.so.8
| ERROR: libcryptopp-8.2.0+gitAUTOINC+9dcc26c582-r8 do_package: Error executing a python function in exec_python_func() autogenerated:
| [...]
| Exception: Exception: KeyError: 'getpwuid(): uid not found: 1001'
| Path ./package/usr/lib/libcryptopp.so.8 is owned by uid 1001, gid 1001, which doesn't match any user/group on target. This may be due to host contamination.
| [...]
| ERROR: Task (/home/thomas/yocto/meta-X2/recipes-X2/libcryptopp/libcryptopp_8.2.0.bb:do_package) failed with exit code '1'
`------------------

I checked the list and found similar problems but not a real solution for me.

libcryptopp's Makefile uses "cp" for installation. I understand this is bad so I patched the Makefile to use "install" instead. That is the recommended usage I could read in the yocto manual. But this does not help either.

As I understand will the do_install and oe_runmake using the bitbake user's UID for installation in a temp dir. Afterwards the do_package uses fakeroot to create the package with root:root file ownership in the package. Right?
In this case I don't understand where the problem is.

I do not run bitbake as root - why should I? So I don't see a possibility to change the files in the temp dir with "chmod" or "install -o 0".

My current bitbake user has UID 1001. On the old server I think it was 1000. Can this be a problem? I tried a new user with UID 1111 but got the same error. Unfortunatelly I can't test with user 1000. It is occupied.

I am out of ideas. Can someone please poke me in the right direction?

Thanks a lot!
Tom


[-- Attachment #2: libcryptopp_8.2.0.bb --]
[-- Type: application/octet-stream, Size: 1429 bytes --]

SUMMARY = "A free C++ class library of cryptographic schemes"

DESCRIPTION = "Free C++ library for cryptographic schemes written by \
Wei Dai and includes ciphers, message authentication codes, one-way \
hash functions, public-key cryptosystems, key agreement schemes, and \
deflate compression."

HOMEPAGE = "https://www.cryptopp.com/"
BUGTRACKER = "https://github.com/weidai11/cryptopp/issues"
SECTION = "libs"

LICENSE = "PD"
LIC_FILES_CHKSUM = "file://License.txt;md5=ec6fbb43f1344336772a925171d9f358"

BBCLASSEXTEND = "native nativesdk"

SRCREV = "9dcc26c58213abb8351fbb1b2a7a1d2c667366e4"

# PR - The revision of this recipe.
PR = "r8"

PV = "8.2.0+git${SRCPV}"

S = "${WORKDIR}/git"

SRC_URI = "git://github.com/weidai11/cryptopp.git \
    file://0001_libcryptopp_8.2.0_do-not-overwrite-CC.patch \
    file://0002_libcryptopp_8.2.0_use-install-instead-of-cp.patch"

EXTRA_OECONF = "--libdir=${base_libdir}"

PACKAGES = "${PN}-dbg ${PN}-dev ${PN}"
FILES_${PN}-dev = "${includedir}/cryptopp/"
FILES_${PN} = "${libdir}/"
FILES_${PN}-dbg = "${libdir}/.debug/ ${prefix}/src/debug/"

INSANE_SKIP_${PN} += "dev-so staticdev"
INSANE_SKIP_${PN}-dev += "dev-so staticdev"
INSANE_SKIP_${PN}-dbg += "dev-so staticdev"

do_compile() {
    oe_runmake static
    oe_runmake dynamic
    oe_runmake libcryptopp.pc 'PREFIX=${prefix}'
}

do_install_prepend() {
    export PREFIX=${D}/${prefix}
}

do_install() {
    oe_runmake install-lib
}

^ permalink raw reply	[flat|nested] only message in thread
only message in thread, other threads:[~2021-05-07  8:34 UTC | newest]

Thread overview: (only message) (download: mbox.gz / follow: Atom feed)
-- links below jump to the message on this page --
2021-05-07  8:34 Path ./package/usr/lib/libcryptopp.so.8 is owned by uid 1111, gid 1111, which doesn't match any user/group on target. This may be due to host contamination Thomas Hill

